libs/utils/src/sync/heavier_once_cell.rs

```diff
@@ -192,6 +192,14 @@ impl<T> OnceCell<T> {
         }
     }
 
+    /// Like [`Guard::take_and_deinit`], but will return `None` if this OnceCell was never
+    /// initialized.
+    pub fn take_and_deinit(&mut self) -> Option<(T, InitPermit)> {
+        let inner = self.inner.get_mut().unwrap();
```

Comment:
Shouldn't this be a `?` here if you want it to match what the rustdoc says?

Reply:
`Inner::take_and_deinit` returns an Option, but perhaps I should demonstrate that.

Reply:
Added a test case in d32f9f4fabd57908ecbaeb6e00e3014d1c24a036 which revealed that the `Inner::take_and_deinit` had a bug in the case where value was uninitialized. It would not had interfered with calls made from `LayerInner::drop` as the result was only semaphore permit exhaustion and after `LayerInner::drop` there are no more initialization or any access to `LayerInner::inner`.

Fixed in 75cc1aea40915952a00d9c7814e90efdef3e059f, refactored the method in 70bd302472bc8d811d1e2afe1b27c9d4b7383090.

Reply:
Above were last three commits, needed to rebase to avoid a test-report problem.